build-scripts/package-tests.cake

```diff
@@ -0,0 +1,382 @@
+                Warnings = 0,
+                Inconclusive = 2 * 1,
+                Skipped = 2 * 7
+            },
+            NUnitProjectLoader));
+    }
+
+    //// Level 2 tests are run for PRs and when packages will be published
+
+    //PackageTests.Add(new PackageTest(2, "Run mock-assembly.dll built for NUnit V2",
+    //    "v2-tests/mock-assembly.dll",
+    //    new ExpectedResult("Failed")
+    //    {
+    //        Total = 28,
+    //        Passed = 18,
+    //        Failed = 5,
+    //        Warnings = 0,
+    //        Inconclusive = 1,
+    //        Skipped = 4
+    //    },
+    //    NUnitV2Driver));
+}
+
+public abstract class NetCorePackageTester : PackageTester
+{
+    public NetCorePackageTester(BuildParameters parameters) : base(parameters)
+    {
+        PackageTests.Add(new PackageTest(1, "Run mock-assembly targeting .NET Core 2.1",
+            "netcoreapp2.1/mock-assembly.dll",
+            new ExpectedResult("Failed")
+            {
+                Total = 37,
+                Passed = 23,
+                Failed = 5,
+                Warnings = 0,
+                Inconclusive = 1,
+                Skipped = 7
+            }));
+
+        PackageTests.Add(new PackageTest(1, "Run mock-assembly targeting .NET Core 3.1",
+            "netcoreapp3.1/mock-assembly.dll",
+            new ExpectedResult("Failed")
+            {
+                Total = 37,
+                Passed = 23,
+                Failed = 5,
+                Warnings = 0,
+                Inconclusive = 1,
+                Skipped = 7
+            }));
+
+        PackageTests.Add(new PackageTest(1, "Run both copies of mock-assembly together",
+            "netcoreapp2.1/mock-assembly.dll netcoreapp3.1/mock-assembly.dll",
+            new ExpectedResult("Failed")
+            {
+                Total = 2 * 37,
+                Passed = 2 * 23,
+                Failed = 2 * 5,
+                Warnings = 0,
+                Inconclusive = 2 * 1,
+                Skipped = 2 * 7
+            }));
+
+        // TODO: Should we have a .NET 5.0 test?
+        //PackageTests.Add(new PackageTest(1, "Run mock-assembly.dll under .NET 5.0",
+        //    "net5.0/mock-assembly.dll",
+        //    new ExpectedResult("Failed")
+        //    {
+        //        Total = 32,
+        //        Passed = 19,
+        //        Failed = 5,
+        //        Warnings = 0,
+        //        Inconclusive = 1,
+        //        Skipped = 7
+        //    }));
+    }
+}
+
+public class ZipPackageTester : NetFXPackageTester
+{
+    public ZipPackageTester(BuildParameters parameters) : base(parameters) { }
+
+    protected override string PackageName => _parameters.ZipPackageName;
+    protected override FilePath PackageUnderTest => _parameters.ZipPackage;
+    protected override string PackageTestDirectory => _parameters.ZipTestDirectory;
+    protected override string PackageTestBinDirectory => PackageTestDirectory + "bin/net20/";
```

Comment:
What's PackageTestBinDirectory meant to be sorry, as opposed to PackageTestDirectory? Is the former the location of the console, and the latter the "root" of the package? Could we maybe add some comments to these variables - or perhaps there's a more descriptive way we could name them?